kernel/arch/x86_64/cpu.cpp

```diff
@@ -135,14 +145,41 @@ void __cpu_identify(void)
      */
     if ((cpu.family == 0xf && cpu.model > 0x2) || (cpu.family == 0x6 && cpu.model >= 0xe))
         cpu.constant_tsc = true;
-#if 0
-	/* TODO: Add 15h support */
-	if(__get_cpuid(0x15, &eax, &ebx, &ecx, &edx))
-	{
-		INFO("x86cpu", "0x15 supported!\n");
-		halt();
-	}
-#endif
+
+    cpu.apic_rate = 0;
+    cpu.tsc_rate = 0;
+    if (__get_cpuid(0x15, &eax, &ebx, &ecx, &edx))
+    {
+        if (ebx == 0 || eax == 0)
+        {
+            INFO("x86cpu", "CPUID 0x15 is useless, reverting to calibration\n");
```

Comment:
Same single-brace statement (apply to elsewhere in the review)
Don't use INFO but pr_info (maybe pr_info("x86: ..."))

Reply:
I think it would be better to get rid of INFO in this and other files in a separate PR

Reply:
I'm not asking you to get rid of all INFOs, just don't add new ones